linked_list/linked_list.py

```diff
@@ -84,13 +121,24 @@ def visit(self):
         
         print(", ".join(helper_list))
 
-    # method to reverse the singly linked list
-    # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    def reverse_helper(self, current):
+        if current.next.next is None:
+            self.head = current.next
+            current.next.next = current
+            return
+
+        self.reverse_helper(current.next)
+        current.next.next = current
+        return current
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def reverse(self):

```

Comment:
👍 Neat recursive solution.  Just a note that since the helper is recursive the call stack is O(n) in size since you basically recur once for each node.  This causes a space complexity of O(n).